### ISSUE TITLE
New Rule: disallow concat for innerHTML - use insertAdjacentHTML instead

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#16215]
### Rule details

disallow appending to innerHTML

### Related ECMAScript feature

Element.innerHTML

### What type of rule is this?

Warns about a potential problem

### Example code

```js
myElement.innerHTML += mycode;
myElement.innerHTML += 'hello';
myElement.innerHTML = myElement.innerHTML + mycode;
myElement.innerHTML = myElement.innerHTML + 'hello';
myElement.innerHTML = someDomElement.innerHTML.concat(mycode);
myElement.innerHTML = someDomElement.innerHTML.concat('hello');
```

### Why should this rule be in the core instead of a plugin?

Basic operation

Concat with .innerHTML re-adds all existing HTML and leads to removing all the listeners attached to the re-added elements.
Use https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML instead.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

**Comment by nzakas** [comment]
Thanks for the suggestion. The ESLint core rules are meant to be runtime agnostic, and because this rule would only apply to a browser runtime, it’s not a candidate for a core rule. You can definitely still create a custom rule to do this, though.
